integr8ly/integreatly-operator
# Watch users and add to rhmi-developers group

Jira: https://issues.jboss.org/browse/INTLY-2836

- All users in an integreatly cluster should be part of the `rhmi-developers` group.
- integreatly-operator should create the `rhmi-developers` group if it isn't there and update the groups Users list accordingly.
- Users should be able to be added and removed from the `dedicated-admins` group and get escalated privileges.


### Verification
1. Deploy integreatly-operator using my OperatorSource
```yaml
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: integreatly-operators
  namespace: openshift-marketplace
spec:
  authorizationToken: {}
  displayName: Integreatly Operators
  endpoint: 'https://quay.io/cnr'
  publisher: Integreatly Publisher
  registryNamespace: sedroche
  type: appregistry
```
This should create the `rhmi-developers` group and add any users to the group.
```bash
$ oc describe group rhmi-developers
Name:           rhmi-developers
Created:        22 minutes ago
Labels:         <none>
Annotations:    <none>
Users:          opentlc-mgr
```
2. Add an extra user to the cluster if you don't have one
```bash
htpasswd -c -b htpasswd testuser Password1
```
and add that user/pass into the htpasswd secret in openshift-config namespace

3. Create a namespace `test1`
4. Logout and login with testuser
5. Check permissions can be added and removed to/from the groups

```bash
$ oc adm policy add-role-to-group view rhmi-developers -n test1
// testuser should see the test1 namespace but not have create perms

$  oc adm groups new dedicated-admins
$ oc adm policy add-role-to-group edit dedicated-admins -n test1
// dedicated-admins should be have create perms in that namespace

$ oc adm groups add-users dedicated-admins testuser
// testuser should have create perms on the namespace

$ oc adm groups remove-users dedicated-admins testuser
// testuser should revert back to just view perms
```
